twitter.js

```diff
@@ -49,9 +61,69 @@ class Twitter {
     };
 
     this.url = getUrl(config.subdomain);
+    this.oauth = getUrl(config.subdomain, "oauth");
     this.config = config;
   }
+  
+  async getRequestToken(twitterCallbackUrl) {
+    // return `oauth_token` & `oauth_token_secret`
+    // https://developer.twitter.com/en/docs/basics/authentication/api-reference/request_token
+    
+    const requestData = {
+      url: `${this.oauth}/request_token`,
+      method: "POST"
+    };
+    
+    var parameters = {};
+    if(twitterCallbackUrl) parameters = { "oauth_callback": twitterCallbackUrl };
+    if (parameters) requestData.url += "?" + querystring.stringify(parameters);
+
+    let headers = {};
+    headers = this.client.toHeader(
+      this.client.authorize(requestData, {})
+    );
+    
+    const results = await Fetch(requestData.url, {
+      method: "POST",
+      headers: Object.assign({}, baseHeaders, headers)
+    })
+    .then(res => res.text())
+    .then(txt => formatOauthRes(txt));
+    
+    return results;
+  }
+  
+  async getAccessToken(options) {
+    // { key:reqTkn, secret:reqTknSecret, verifier:oauthVerifier }
+    // return `accTkn_key` & `accTkn_secret`
+    // https://developer.twitter.com/en/docs/basics/authentication/api-reference/access_token
+    
+    const requestData = {
+      url: `${this.oauth}/access_token`,
+      method: "POST"
+    };
+    
+    var parameters = { "oauth_verifier": options.verifier };
+    if (parameters) requestData.url += "?" + querystring.stringify(parameters);
 
+    let headers = {};
+    headers = this.client.toHeader(
+      this.client.authorize(requestData, {
+        key: options.key,
+        secret: options.secret
+      })
+    );
+
+    const results = await Fetch(requestData.url, {
+      method: "POST",
+      headers: Object.assign({}, baseHeaders, headers)
+    })
+    .then(res => res.text())
+    .then(txt => formatOauthRes(txt));
```

Comment:
This could be done with `txt => querystring.parse(txt.split('?')[1])` I think. Does twitter not return JSON for this endpoint? It'd be nice to not have to parse the text at all.

Reply:
And the same comments apply for the `getAccessToken` method.

Reply:
unfortunately, according to the docs, the result is returned in the form of a url-encoded string :"(